frontend/stmt_inferrer.py

```diff
@@ -331,10 +332,46 @@ def _infer_class_def(node, context, solver):
 
     class_attrs = solver.z3_types.instance_attributes[node.name]
     instance_type = solver.z3_types.classes[node.name]
+    class_to_funcs = solver.z3_types.class_to_funcs[node.name]
+    base_class_to_funcs = {}
+    base_attrs = {}
+    if node.bases and isinstance(node.bases[0], ast.Name):
+        base_class_to_funcs = solver.z3_types.class_to_funcs[node.bases[0].id]
+        base_attrs = solver.z3_types.instance_attributes[node.bases[0].id]
 
     for attr in class_context.types_map:
         solver.add(class_attrs[attr] == class_context.types_map[attr],
                    fail_message="Class attribute in {}".format(node.lineno))
+        if attr in base_class_to_funcs:
+            # class_to_funcs[attr] is a tuple of two elements.
+            # The first is the args length, the second is the decorators list
+            base_args_len = base_class_to_funcs[attr][0]
+            sub_args_len = class_to_funcs[attr][0]
+            decorators = base_class_to_funcs[attr][1] + class_to_funcs[attr][1]
+
+            if not base_args_len:
+                raise TypeError("Class methods should have at least one argument (the receiver)."
```

Comment:
Wait, when you say "Class methods", do you mean normal methods in classes (that are supposed to have a self parameter)? In that case you should use a different term, since Python also has a @classmethod decorator that means something different-